android/src/main/java/cl/json/ShareFile.java

```diff
@@ -117,17 +112,17 @@ private String getRealPathFromURI(Uri contentUri) {
     public Uri getURI() {
 
         final MimeTypeMap mime = MimeTypeMap.getSingleton();
-        this.extension = mime.getExtensionFromMimeType(getType());
+        String extension = mime.getExtensionFromMimeType(getType());
         final String authority = ((ShareApplication) reactContext.getApplicationContext()).getFileProviderAuthority();
 
         if(this.isBase64File()) {
             String encodedImg = this.uri.getSchemeSpecificPart().substring(this.uri.getSchemeSpecificPart().indexOf(";base64,") + 8);
             try {
                 File dir = new File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_DOWNLOADS );
-                if (!dir.exists()) {
-                    dir.mkdirs();
```

Comment:
Maybe it is better to keep this?

Reply:
More than keep it - I altered it to not throw away the return value of "dir.mkdirs()" - now that becomes part of a chain of dir.exists + dir.mkdirs, and if one or the other of them doesn't work we get notice. Previously if dir.mkdirs() returned false we'd never know

Reply:
Nice approach, thanks